projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.ts

```diff
@@ -0,0 +1,154 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {BaseValueComponent} from '../base-value.component';
+import {CreateLinkValue, ReadLinkValue, ReadResource, UpdateLinkValue, KnoraApiConnection} from '@knora/api';
+import {Subscription} from 'rxjs';
+import {AbstractControl, FormBuilder, FormControl, FormGroup, ValidatorFn} from '@angular/forms';
+import {KnoraApiConnectionToken} from '../../../core';
+
+export function resourceValidator(): ValidatorFn {
+  return (control: AbstractControl): { [key: string]: any } | null => {
+    const invalid = !(control.value instanceof ReadResource);
+    return invalid ? {'invalidType': {value: control.value}} : null;
+  };
+}
+
+@Component({
+  selector: 'kui-link-value',
+  templateUrl: './link-value.component.html',
+  styleUrls: ['./link-value.component.scss']
+})
+
+export class LinkValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() displayValue?: ReadLinkValue;
+  @Input() parentResource: ReadResource;
+  @Input() propIri: string;
+  resources: ReadResource[] = [];
```

Comment:
@tobiasschweizer I have already initialized it here to an empty array

Reply:
I understood that your idea of clearing selection empties the list when the user types and the string is still too short. I think this idea is good.

Reply:
I think this is important when user selects a resource and then wants to change it again. Then the selection has to change to.

Reply:
because I think otherwise you still get the old selection when the label string is shorter than 3 chars

Reply:
but maybe I am wrong, I will leave this to you. Please proceed as you want.